### ISSUE TITLE
Empty RuleParameter is required if there is a rule has undefined identifier in Expression

### ISSUE THREAD

**Issue by kamapcrazy** [microsoft/RulesEngine#38]
Step to reproduce:

- Here is the workflow rules:
  - *Noted that the second rule "AvailableIf" contains two more expression identifiers: productStock and customer*
```
[
  {
    "WorkflowName": "DisplayProductAvailability",
    "WorkflowRulesToInject": null,
    "Rules": [
      {
        "RuleName": "AlwaysAvailable",
        "Operator": null,
        "ErrorMessage": "test",
        "ErrorType": "Warning",
        "RuleExpressionType": "LambdaExpression",
        "WorkflowRulesToInject": null,
        "Rules": null,
        "LocalParams": null,
        "Expression": "input.Name == \"Normal product\"",
        "SuccessEvent": null
      },
      {
        "RuleName": "AvailableIf",
        "Operator": "AndAlso",
        "ErrorMessage": "Not Available",
        "ErrorType": "Warning",
        "RuleExpressionType": null,
        "WorkflowRulesToInject": null,
        "Rules": [
          {
            "RuleName": "InStock",
            "Operator": null,
            "ErrorMessage": "Out of stock",
            "ErrorType": "Warning",
            "RuleExpressionType": "LambdaExpression",
            "WorkflowRulesToInject": null,
            "Rules": null,
            "LocalParams": null,
            "Expression": "input.Quantity < productStock.Quantity",
            "SuccessEvent": null
          },
          {
            "RuleName": "VisibleToCustomer",
            "Operator": null,
            "ErrorMessage": "Product is not available at customer location",
            "ErrorType": "Warning",
            "RuleExpressionType": "LambdaExpression",
            "WorkflowRulesToInject": null,
            "Rules": null,
            "LocalParams": null,
            "Expression": "productStock.Location == customer.Location",
            "SuccessEvent": null
          }
        ],
        "LocalParams": null,
        "Expression": null,
        "SuccessEvent": "In stock and visible to customer"
      }
    ]
  }
]
```

- Execute the workflow with one RuleParameter:
```
dynamic input = new ExpandoObject();
input.Name = "Normal product";
var ruleParameters = new[]
   {
      new RuleParameter( nameof( input ), input )
   };
RulesEngine.ExecuteRule("DisplayProductAvailability", ruleParameters );
```
Result: *An exception is thrown: "Target object is not an ExpandoObject"*
**My expection is that the RulesEngine run successfully and the RuleResult should be success for rule name "AlwaysAvailable"**

  - Execute the workflow with two RuleParameters:
```
dynamic input = new ExpandoObject();
input.Name = "Normal product";
var ruleParameters = new[]
   {
      new RuleParameter( "empty", "" ), // empty rule parameter and not related to any rules expressions
      new RuleParameter( nameof( input ), input )
   };
RulesEngine.ExecuteRule("DisplayProductAvailability", ruleParameters );
```
Result: *RulesEngine execute successfully, the first rule name is returned "AlwaysAvailable"*

**Comment by abbasc52** [comment]
@kamapcrazy Can you try v2.1.1 and see if this issue still persists?

**Comment by abbasc52** [comment]
this issue is fixed as a part of 2.1.1, you can refer this test case

https://github.com/microsoft/RulesEngine/blob/6b66162e561b5447c4c3500f167f8823c3c7cbbe/test/RulesEngine.UnitTest/BusinessRuleEngineTest.cs#L220-L234
